test/integration/query-interface.test.js

```diff
@@ -41,6 +41,28 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
     });
   });
 
+  describe('showAllTables', () => {
+    it('should contain only tables', function() {
+      const cleanup = r => {
+        return this.sequelize.query('DROP VIEW V_Fail').then(() => r, () => r);
+      };
+      return this.queryInterface
+        .createTable('my_test_table', { name: DataTypes.STRING })
+        .then(cleanup)
+        .then(() => this.sequelize.query('CREATE VIEW V_Fail AS SELECT 1 Id'))
+        .then(() => this.queryInterface.showAllTables())
+        .then(cleanup)
+        .then(tableNames => {
+          if (tableNames[0] && tableNames[0].tableName) {
+            tableNames = tableNames.map(v => v.tableName);
+          }
+          expect(tableNames).to.have.length(1);
+          expect(tableNames).to.contain('my_test_table');
+          expect(tableNames).to.not.contain('V_Fail');
```

Comment:
Since you are asserting that the array has length 1 and asserting that it contains `'my_test_table'`, automatically it won't contain `'V_Fail`, right? I think you should refactor these lines to a simpler assert:

```js
expect(tableNames).to.deep.equal(['my_test_table']);
```